### DISCUSSION TITLE
listView duplicates values

### DISCUSSION THREAD

**Discussion by crawe1** [YetiForceCompany/YetiForceCRM#14927] [How to] [answered]
Hi,
i have a questions from listview. 
I want to try add new module with manual ids on static this record import directly to database not by create record but this is not problem. But when i want to display record in module array is ok, a sume from column is ok but in module i see only one of each (last one). i think this is some filter maybe (anti-duplicate) but i dont know where. 
@mariuszkrzaczkowski @vovpff you have any idea why ?

yes - my import data (products have this same ids) 

display view:
![image](https://user-images.githubusercontent.com/61950059/124825492-00f21e80-df74-11eb-842e-f751bacb7b13.png)

array:
Array ( 
'[0] => Array ( [id] => 7650 [subject] => BL RAMA STAL. PIONOWA LW 1,50 X 0,73 M [bazaprod_magazynid] => 115 [bazaprod_ilosctowaru] => 200.00 [bazaprod_kwotajedn] => 144.12000000 [bazaprod_dostawcaid] => 7535 [bazaprod_dataprzyj] => 2021-07-07 [bazaprod_przyjecie] => 12221 [number] => 1700.150 ) 
[1] => Array ( [id] => 7650 [subject] => BL RAMA STAL. PIONOWA LW 1,50 X 0,73 M [bazaprod_magazynid] => 115 [bazaprod_ilosctowaru] => 500.00 [bazaprod_kwotajedn] => 225.18000000 [bazaprod_dostawcaid] => 7586 [bazaprod_dataprzyj] => 2021-07-07 [bazaprod_przyjecie] => 12222 [number] => 1700.150 ) 
[2] => Array ( [id] => 7650 [subject] => BL RAMA STAL. PIONOWA LW 1,50 X 0,73 M [bazaprod_magazynid] => 115 [bazaprod_ilosctowaru] => 300.00 [bazaprod_kwotajedn] => 94.58000000 [bazaprod_dostawcaid] => 7580 [bazaprod_dataprzyj] => 2021-07-06 [bazaprod_przyjecie] => 12224 [number] => 1700.150 ) 
[3] => Array ( [id] => 7661 [subject] => BL ST-RAMA PIONOWA 2.00X0.73m, 4UCHWYTY [bazaprod_magazynid] => 115 [bazaprod_ilosctowaru] => 888.00 [bazaprod_kwotajedn] => 102.80000000 [bazaprod_dostawcaid] => 7573 [bazaprod_dataprzyj] => 2021-07-07 [bazaprod_przyjecie] => 12226 [number] => 1710.200 ) 
[4] => Array ( [id] => 7650 [subject] => BL RAMA STAL. PIONOWA LW 1,50 X 0,73 M [bazaprod_magazynid] => 115 [bazaprod_ilosctowaru] => 777.00 [bazaprod_kwotajedn] => 123.00000000 [bazaprod_dostawcaid] => 7614 [bazaprod_dataprzyj] => 2021-07-07 [bazaprod_przyjecie] => 12227 [number] => 1700.150 ) )'

when i change on module/modulename.php index on another column with different ids in column, everything is ok. 

![image](https://user-images.githubusercontent.com/61950059/124825068-73163380-df73-11eb-9dd0-9e532a58e5f9.png)

**Comment by vovpff** [comment]
Very strange. I don't have any ideas now.

Are you use id as productid?

**Reply by crawe1** [reply]
yes i use this table for table with all external reception, pz/pzk/serviceback/.This is automaticly build table for other modules wz/wzk/crash/serviceout etc. its simple i not add by button records but i put products use on other module to this table and this table link by 'name' to other module and i want to see this position. crmenity resolve names products (litle cheat but working).  But i dont know what merge records this same id in display view. As you see array is ok but only view list not working on my model. 

yes this is product id but as you can see from array on table i have more this same products but different is other data.

**Comment by vovpff** [comment, marked as answer]
Could you share `vtiger_products` table dump after this kind of import?

**Reply by crawe1** [reply]
$vtiger_products = array(
  array('productid' => '7650','product_no' => 'PRO2','productname' => 'BL RAMA STAL. PIONOWA LW 1,50 X 0,73 M','productcode' => '','pscategory' => NULL,'manufacturer' => 'Layher','qty_per_unit' => '0.00','unit_price' => '[]','weight' => '15.800','pack_size' => NULL,'sales_start_date' => NULL,'sales_end_date' => NULL,'start_date' => NULL,'expiry_date' => NULL,'cost_factor' => NULL,'commissionrate' => '0.000','commissionmethod' => NULL,'discontinued' => '1','usageunit' => 'szt.','reorderlevel' => '0','website' => '','mfr_part_no' => '','vendor_part_no' => '1700.150','serialno' => '','qtyinstock' => '200.000','productsheet' => '','qtyindemand' => '0','glacct' => '','vendor_id' => '0','imagename' => '','taxes' => '','ean' => '','subunit' => '','renewable' => '0','category_multipicklist' => NULL,'purchase' => '[]','products_gtu' => 'GTU_08','produkt_pz' => '12229','produkt_dostawca' => '7587','produkt_cena' => '45.0000','produkt_istniejacy' => '0','produkt_pzk' => '0','produkt_wz' => '','produkt_wzk' => '','as' => '0'),


  array('productid' => '7661','product_no' => 'PRO13','productname' => 'BL ST-RAMA PIONOWA 2.00X0.73m, 4UCHWYTY','productcode' => '','pscategory' => NULL,'manufacturer' => 'Layher','qty_per_unit' => '0.00','unit_price' => '{"currencies":{"1":{"price":102.8}},"currencyId":1}','weight' => '19.600','pack_size' => NULL,'sales_start_date' => NULL,'sales_end_date' => NULL,'start_date' => NULL,'expiry_date' => NULL,'cost_factor' => NULL,'commissionrate' => '-99.225','commissionmethod' => NULL,'discontinued' => '1','usageunit' => 'szt.','reorderlevel' => '0','website' => '','mfr_part_no' => '','vendor_part_no' => '1710.200','serialno' => '','qtyinstock' => '888.000','productsheet' => '','qtyindemand' => '0','glacct' => '','vendor_id' => '8772','imagename' => '','taxes' => '','ean' => '','subunit' => '','renewable' => '0','category_multipicklist' => NULL,'purchase' => '{"currencies":{"1":{"price":13261.2}},"currencyId":1}','products_gtu' => 'GTU_08','produkt_pz' => '12226','produkt_dostawca' => '7573','produkt_cena' => '102.8000','produkt_istniejacy' => '0','produkt_pzk' => '0','produkt_wz' => 'WZ/3/01/21, WZ/10/03/21','produkt_wzk' => '','as' => '0')
);

**Reply by vovpff** [reply]
What about mysql dump?

**Reply by crawe1** [reply]
Ok ok resolve problem display view step by step. Thank you @vovpff again 

https://github.com/YetiForceCompany/YetiForceCRM/blob/3b9a4ec2f6f6e8ed6bc2bd12abf0246de59412db/modules/Vtiger/models/ListView.php#L600

**Reply by vovpff** [reply]
Ok. I ask because I don't understand what is going on

**Reply by crawe1** [reply]
:) this is more strange, complicated problem :) @vovpff thank you one more time. !

**Comment by crawe1** [comment]
@vovpff maybe you can help me one more time i want to get data from this searchfields
![image](https://user-images.githubusercontent.com/61950059/125432068-63baba3b-d747-4a15-9714-ab901b9eb6ad.png)
to this place ?
![image](https://user-images.githubusercontent.com/61950059/125432395-55681fe4-f2b5-4a7b-9b0a-d9dc5cf300af.png)
but not by mappfield in database beacouse this is not this same module (module is tmp create table but id is products) :( maybe some functions ?

**Reply by vovpff** [reply]
I know two ways.
1) handler what update this column after saving record
2) paid extension

**Reply by crawe1** [reply]
hmm but how to get select record from search list ? search list -> sent all selected record do module after click ?

**Reply by vovpff** [reply]
For each related entity you need to get value  and put it into table of parent module
